pkg/errors/types.go

```diff
@@ -23,13 +26,13 @@ func (ErrSecretStoreConn) Error() string {
 	return "Unable to obtain from underlying data-store"
 }
 
-// ErrSecretNotFound error when a secret cannot be found. This aids in differentiating between empty("") values and non-existent keys
-type ErrSecretNotFound struct {
-	Key string
+// ErrSecretsNotFound error when a secret cannot be found. This aids in differentiating between empty("") values and non-existent keys
+type ErrSecretsNotFound struct {
+	Keys []string
```

Comment:
Would you be open to making this a private member so that it can't accidentally be reset on the same instance? Set the value via constructor per pattern [found](https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/metadata/errors/types.go) in edgex-go.

Reply:
Good point. I think that is a better approach.